core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1108,6 +1108,25 @@ public Builder setDefaultPartitionSpec(int specId) {
       return this;
     }
 
+    Builder removeSpecIds(Iterable<Integer> specIds) {
```

Comment:
I still feel like this should be called `removeSpecs`. If we look at the other APIs on TableMetadata builder like `setDefaultPartitionSpec(int specId)` it aligns with that naming, I don't think we need the "Ids" suffix.

Reply:
I'm not super opinionated here though, I remember the early discussion we had where I also mentioned removeSpecIds but this is just something I noticed when taking a look just now.